### DISCUSSION TITLE
How to automatically change the name “Name […] of this device”

### DISCUSSION THREAD

**Discussion by silverhook** [piroor/treestyletab#3218] [Q&A]
I rely **heavily** on having several Firefox Profiles and spinning up a new one with my default settings as fast as possible is paramount. (see links below for reference)

As part of that I am trying to automate the naming of the Firefox Sync devices so they match with the KDE Plasma Activity name and as such the Firefox Profile, and it would be great to also do the same for TST, since most of the time I use TST to send tab( tree)s from one Profile/device to another.

I got so far that for this is handled in Firefox by `identity.fxaccounts.account.device.name` in `about:config`…

But where is it handled in TST?

How can I through a shell script or otherwise change this setting in TST?

---

Blog explaining the issue and the 0.1 version of the solution:
https://matija.suklje.name/introducing-activity-aware-firefox

Code repository to the solution, which is at 0.3 now and much improved:
https://gitlab.com/hook/activity-aware-firefox

Relevant issue:
https://gitlab.com/hook/activity-aware-firefox/-/issues/8

**Comment by piroor** [comment]
The most easiest and realistic way is: implementing a native messaging host. It will allow to access any data on the platform from TST.

The top concern is: violation of the project scope. TST project aims to provide tree management feature, and Firefox Sync simulation is just an option. If there is any existing native messaging host providing features to get the device name, I'm OK to add some API call to use the host and request to add TST to the allowed list of the host. But implementing new native messaging host by myself is out of scope.

**Comment by silverhook** [comment]
With the easiest and most realistic way out of the question then, what would be the brute force approach then? ;)

**Reply by piroor** [reply]
TST's options are stored in Firefox's profile via `browser.storage.local()` WebExtensions API.
https://developer.mozilla.org/docs/Mozilla/Add-ons/WebExtensions/API/storage/local
The backend where the stored data is saved to is not stable - on old versions of Firefox it is saved to a JSON file, and currently it is stored to an SQLite DB via IndexedDB API. I don't know which file is the substance of the DB but you can browse its records via `about:debugging` => Tree Style Tab => Inspect => Storage => Extension Storage.
